.github/workflows/deploy.yml

```diff
@@ -0,0 +1,74 @@
+name: Deploy to a Cloud Platform environment
+on:
+  workflow_call:
+    inputs:
+      environment:
+        required: true
+        type: string
+      ECR_REGION:
+        required: true
+        type: string
+      ECR_REPOSITORY:
+        required: true
+        type: string
+    secrets:
+      ECR_ROLE_TO_ASSUME:
+        required: true
+      KUBE_CERT:
+        required: true
+      KUBE_CLUSTER:
+        required: true
+      KUBE_NAMESPACE:
+        required: true
+      KUBE_TOKEN:
+        required: true
+
+
+jobs:
+  deploy:
+    name: Deploy
+    environment: ${{ inputs.environment }}
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write # This is required for requesting the JWT
+      contents: read  # This is required for actions/checkout
+    steps:
+      - name: Checkout GitHub repository
+        uses: actions/checkout@v4
+
+      - name: Authenticate to the cluster
+        env:
+          KUBE_NAMESPACE: ${{ secrets.KUBE_NAMESPACE }}
+          KUBE_CLUSTER: ${{ secrets.KUBE_CLUSTER }}
+        run: |
+          echo "${{ secrets.KUBE_CERT }}" > ca.crt
+          kubectl config set-cluster ${KUBE_CLUSTER} --certificate-authority=./ca.crt --server=https://${KUBE_CLUSTER}
+          kubectl config set-credentials deploy-user --token=${{ secrets.KUBE_TOKEN }}
+          kubectl config set-context ${KUBE_CLUSTER} --cluster=${KUBE_CLUSTER} --user=deploy-user --namespace=${KUBE_NAMESPACE}
+          kubectl config use-context ${KUBE_CLUSTER}
+
+      - name: Assume role in Cloud Platform
```

Comment:
Might be worth checking these two bits, I think I normally add ECR registry as variable and not require login for deployment.

Reply:
Thanks Jav, great point! I have moved the ECR Registry URL to a repository secret with this commit: [https://github.com/ministryofjustice/laa-access-civil-legal-aid/pull/3/commits/9024ed52c39ce0f99ad8be4fba416db987efe8af](https://github.com/ministryofjustice/laa-access-civil-legal-aid/pull/3/commits/9024ed52c39ce0f99ad8be4fba416db987efe8af)